### PR TITLE
extended code to allow L > 1 for sampling z in the decoder as defined…

### PR DESCRIPTION
**Pull request by JohannesHeidecke**
… in the paper.

The paper does state that L=1 is usually sufficient, if the batch size is large enough, but with this change it is free to the user to select L.

### CODE REVIEW

File: VAE.py [hunk 1]
```diff
@@ -113,10 +116,10 @@ def decoder(self, x, z):
             log_sigma_decoder = T.dot(h_decoder, self.params['W_hxsigma']) + self.params['b_hxsigma']
 
             logpxz = (-(0.5 * np.log(2 * np.pi) + 0.5 * log_sigma_decoder) -
-                      0.5 * ((x - reconstructed_x)**2 / T.exp(log_sigma_decoder))).sum(axis=1)
+                      0.5 * ((x - reconstructed_x)**2 / T.exp(log_sigma_decoder))).sum(axis=2).sum(axis=0) / self.L
```
Comment (y0ast) [inline VAE.py]: can you turn this into a .mean(axis=0)? avoids the explicit division

File: VAE.py [hunk 2]
```diff
@@ -113,10 +116,10 @@ def decoder(self, x, z):
...
         else:
             reconstructed_x = T.nnet.sigmoid(T.dot(h_decoder, self.params['W_hx']) + self.params['b_hx'].dimshuffle('x', 0))
-            logpxz = - T.nnet.binary_crossentropy(reconstructed_x, x).sum(axis=1)
+            logpxz = - T.nnet.binary_crossentropy(reconstructed_x, x).sum(axis=2).sum(axis=0) / self.L
```
Comment (y0ast) [inline VAE.py]: See above

**Review by y0ast — CHANGES_REQUESTED** [review]
Looks good, could you address the minor things?

File: VAE.py (hunk 2 above)
Comment (JohannesHeidecke) [reply VAE.py]: good point, updated the code!

**Comment by y0ast** [comment]
There are now two dots between the sum and the mean?

**Comment by JohannesHeidecke** [comment]
yup, my bad. fixed it.

**Comment by y0ast** [comment]
Thanks, I'll run some checks in the weekend to verify that it works

**Review by y0ast — APPROVED** [review]

**Comment by y0ast** [comment]
Works well, thanks!

**Comment by JohannesHeidecke** [comment]
🎉

**Pull request merged**